### PR TITLE
Add FuncX client to Garden

### PR DESCRIPTION
**Pull request by WillEngler**
First part of #39.

## Overview

This PR adds a FuncX client to the Garden client. I'm putting this first part up for PR separately so that Owen doesn't have to redo this bit or wait on me if he gets to pipeline registration soon.

## Discussion

This is hopefully pretty by the books. Just following Container Service docs and adapting some things that DLHub does.

## Testing

I exercised the FuncX client with some quick and dirty code that makes a container. (Currently growing that into the next PR.)

----
:books: Documentation preview :books:: https://garden-ai--87.org.readthedocs.build/en/87/

### CODE REVIEW

File: garden_ai/client.py [hunk 1]
```diff
@@ -100,13 +108,23 @@ def __init__(
...
         _request_header_provider_registry.register(BinaryContentTypeHeaderProvider)
 
+    def _make_funcx_client(self):
+        scope_to_authorizer = {
+            AuthScopes.openid: self.openid_authorizer,
+            SearchScopes.all: self.search_authorizer,
+            FuncXClient.FUNCX_SCOPE: self.funcx_authorizer,
+        }
+        funcx_login_manager = FuncXLoginManager(scope_to_authorizer)
+        return FuncXClient(login_manager=funcx_login_manager, do_version_check=False)
```
Comment (WillEngler) [inline garden_ai/client.py:126]: `do_version_check=False` is a bit of a hack. Turning that off means that there are no outbound network calls on FuncXClient init, and thus nothing we need to mock right now. But if there's any pushback on this I'll go back and do the mocking.

Comment (OwenPriceSkelly) [reply garden_ai/client.py:126]: seems fine to me

File: garden_ai/globus_compute/login_manager.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        app_name: Union[str, None] = None,
+    ) -> FuncxWebClient:
+        return FuncxWebClient(
+            base_url=base_url,
+            app_name=app_name,
+            authorizer=self.authorizers[FuncXClient.FUNCX_SCOPE],
+        )
+
+    def ensure_logged_in(self):
+        raise NotImplementedError("ensure_logged_in cannot be invoked from here")
```
Comment (OwenPriceSkelly) [inline garden_ai/globus_compute/login_manager.py:41]: 👍

**Review by OwenPriceSkelly — APPROVED** [review]
lgtm!

**Pull request merged**